CRM/Member/Form/MembershipView.php

```diff
@@ -154,8 +154,21 @@ public function preProcess() {
     $this->assign('context', $context);
 
     if ($this->membershipID) {
-      $params = ['id' => $this->membershipID];
-      CRM_Member_BAO_Membership::retrieve($params, $values);
+      $values = \Civi\Api4\Membership::get()
+        ->addSelect('*', 'status_id:label', 'membership_type_id:label', 'membership_type_id.financial_type_id')
+        ->addWhere('id', '=', $this->membershipID)
+        ->execute()
+        ->first();
+
+      // Ensure keys expected by MembershipView.tpl are set correctly
+      // Some of these defaults are overwritten dependant on context below
+      $values['financialTypeId'] = $values['membership_type_id.financial_type_id'];
+      $values['membership_type'] = $values['membership_type_id:label'];
+      $values['status'] = $values['status_id:label'];
+      $values['owner_contact_id'] = FALSE;
+      $values['owner_display_name'] = FALSE;
+      $values['campaign'] = FALSE;
```

Comment:
One value missing from here that is returned by ` CRM_Member_BAO_Membership::retrieve()` is `"active"` which I think you'd get by adding `"status_id.is_current_member"` to the `select1`.

Reply:
I don't think the template is actualy using `active`, which is why I left it off. That said, it could be used by third-parties if they've overwritten or extended the template, so it doesn't hurt to add it in. I've pushed an update.